corehq/apps/linked_domain/views.py

```diff
@@ -503,17 +483,25 @@ def link_domains(couch_user, upstream_domain, downstream_domain):
 
 def validate_push(user, domain, downstream_domains):
     if not downstream_domains:
-        raise NoDownstreamDomainsProvided
+        raise InvalidPushException(
+            message=gettext("No downstream project spaces were selected. Please contact support.")
+        )
 
-    try:
-        domain_links = [
-            DomainLink.objects.get(master_domain=domain, linked_domain=dd) for dd in downstream_domains
-        ]
-    except DomainLink.DoesNotExist:
-        raise DomainLinkNotFound
+    domain_links = []
+    for dd in downstream_domains:
+        try:
+            domain_links.append(DomainLink.objects.get(master_domain=domain, linked_domain=dd))
+        except DomainLink.DoesNotExist:
+            raise InvalidPushException(
+                message=gettext(
+                    "The project space link between {} and {} does not exist. Ensure the link was not recently "
+                    "deleted.").format(domain, dd)
+            )
```

Comment:
This seems minor and probably unlikely, but the code here stops at the first missing link, which can create that frustrating experience where you fix the first issue, try again, and get the exact same issue when the next item is missing. If this seems like a realistic situation, it might be worthwhile to capture all missing links and collect them into a single message

Reply:
Yeah I thought about this, though in this case I can't imagine any other scenario besides the user just needing to refresh the page, which would resolve the issue entirely even if there were other domain links impacted. The downside is they would not be made aware of those other links, but I assume the user who deleted the links did so for good reason.